docs/cndb_omf.md

```diff
@@ -2,122 +2,158 @@ DRAFT!!! DRAFT!!! DRAFT!!! DRAFT!!! DRAFT!!! DRAFT!!!
 - Seqno: KVDB seqno at the time of logging this record.
-- Ingest id: Every ingest operation gets a unique id. If TX does not describe an ingest operation, the
-  value is set to `CNDB_INVAL_INGESTID`.
-- Transaction horizon: Transaction horizon for an ingest. This is used by WAL to reclaim its files. When
-  the operation is not an ingest, the value is set to `CNDB_INVAL_HORIZON`.
+- Ingest id: Every ingest operation gets a unique id. If `TX_START` does not describe an ingest operation,
+  the value is set to `CNDB_INVAL_INGESTID`.
+- Transaction horizon: Transaction horizon for an ingest. This is used by WAL to reclaim its files.
+  When the operation is not an ingest, the value is set to `CNDB_INVAL_HORIZON`.
 
-### TXNODE
-- TX id: Transaction id to tie this record to the corresponding TX record.
+### TX_NODE
+
+- TX id: Transaction id to tie this record to the corresponding `TX_START` record.
 - Cnid: (CN id) A unique identifier for the KVS.
 - Old node cnt: Number of nodes being deleted
-- New node cnt: Number of nodes being created that will replace the old nodes
-Old Node Ids (packed u64 fields)
-Old Node Ids (packed u64 fields)
+- New node cnt: Number of nodes being created that will replace the old nodes.
+Old Node Ids (packed u64 fields).
+New Node Ids (packed u64 fields).
 
-Node id `node-00` is reserved for the root node. There's no `txnode` record for the root node. A
-root node always exists. TXC records would use `node-00` when a kvset is created in the root node.
+Node id 0 is reserved for the root node. There's no `txnode` record for the root node. A
+root node always exists. `TX_KVSET_CREATE` records would use node id 0 when a kvset is created in the root node.
 
 Only nodes at level 1 are logged explicitly in CNDB.
 
-### TXC
-- TX id: Transaction id to tie this record to the corresponding TX record.
+### TX_KVSET_CREATE
+
+When a kvset is split, unless a kblock is rewritten, the vblock idx stored will not be sufficient to
+fetch the value for a key. In addition to the vblock idx, a vgroup id is also stored in the kmd
+region for the key's value. So the vblock idx acts as the index within a vgroup. After a split, in
+the kvset that's moved to the other node, each vblock index in a vgroup is now offset by a the
+number of vblocks that were moved to the other kvset. The `vgroup->offset` mapping in the
+`TX_KVSET_CREATE` record handles this change.
+
+During a kvset split, some mblocks are rewritten while others are just moved around. If the KVDB
+crashes while there's an ongoing split operation, there's some recovery needed when the KVDB is
+brought back up. This would involve cleaning up the mblocks that were created for this split but not
+the mblocks that were moved. The KMap and VMap bitmaps represent blocks that were created.
+i.e. for a KMap/VMap entry:
+  0: mblock was moved. Do not delete on rollback
+  1: mblock was created. Delete on rollback.
```

Comment:
If the KVDB crashes before the split completes then recovery is required before the KVDB can brough online.  This includes cleaning up the mblocks...